pyquil/quilatom.py

```diff
@@ -35,24 +37,39 @@ def __init__(self, index):
     def out(self):
         return str(self.index)
 
+    def __str__(self):
+        return str(self.index)
+
     def __repr__(self):
         return "<Qubit {0}>".format(self.index)
 
+    def __hash__(self):
+        return hash(self.index)
+
+    def __eq__(self, other):
+        return isinstance(other, Qubit) and other.index == self.index
 
-class QubitPlaceholder(Qubit):
-    def __init__(self):
-        pass
 
-    @property
-    def index(self):
-        raise RuntimeError("Qubit has not been assigned an index")
+class QubitPlaceholder(QuilAtom):
+    def out(self):
+        raise RuntimeError("Qubit {} has not been assigned an index".format(self))
```

Comment:
I would have expected a list of QubitPlaceholders.

Reply:
And `.get_qubits()`

Reply:
>  Something like "Cannot print program, program contains unassigned qubit."

Is the current wording not sufficient?

Reply:
Added changes to support `get_qubits` for placeholders

Added changes to support `str(program)` with qubit placeholders. `program.out()` will still raise an error. The distinction being anything from `.out()` should be ready to run

Reply:
Cool! Thanks @mpharrigan. The wording is probably fine - when I got that error it looked like I was triggering some weird internal thing. I was suggesting that maybe giving a direct message saying that "the reason this programming isn't printing is because you didn't assign the qubits to number".

But that's not necessary. Thanks for adding support for those two things! This PR is awesome.